test_scripts/TheSameApp/Security/001_Send_RPC_or_encripted_RPC_in_protected_unprotected_mode_apps_with_same_appName_appID_from_2_devices.lua

```diff
@@ -0,0 +1,90 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal:
+-- https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0204-same-app-from-multiple-devices.md
+-- Description:
+-- Send encrypted or unencrypted RPCs in protected and unprotected mode from apps registered on different
+-- mobile devices and having same appNames and same appIds
+--
+-- Preconditions:
+-- 1) SDL has up-to-date certificates in Policy Table
+-- 2) App_1 is registered from Mobile №1 and RPC service is started in unprotected mode
+-- 3) App_2 is registered from Mobile №2 and RPC service is started in unprotected mode
+--
+-- Steps:
+-- 1) App_1 switches RPC service to protected mode:
+--   Check:
+--    SDL started protected mode successfully for App1.
+-- 2) Mobile №1 sends AddCommand RPC in protected mode to SDL
+--   Check:
+--    SDL responds with  AddCommand secure response (protected) to Mobile №1
+-- 3) Mobile №1 sends AddCommand RPC in unprotected mode to SDL
+--   Check:
+--    SDL responds with  AddCommand secure response (protected) to Mobile №1
+-- 4) Mobile №2 sends AddCommand RPC in unprotected mode to SDL
+--   Check:
+--    SDL responds with AddCommand response (unprotected) to Mobile №2
+-- 5) App_2 switches RPC service to protected mode:
+--   Check:
+--    SDL started protected mode successfully for App2.
+-- 6) Mobile №2 sends AddCommand RPC in protected mode to SDL
+--   Check:
+--    SDL responds with  AddCommand secure response (protected) to Mobile №2
+-- 7) Mobile №2 sends AddCommand RPC in unprotected mode to SDL
+--   Check:
+--    SDL responds with  AddCommand secure response (protected) to Mobile №2
+-- 8) Mobile №1 sends AddCommand RPC in protected mode to SDL
+--   Check:
+--    SDL responds with  AddCommand secure response (protected) to Mobile №1
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/TheSameApp/Security/commonSecurity')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Data ]]
+local devices = {
+  [1] = { host = "1.0.0.1",         port = config.mobilePort },
+  [2] = { host = "192.168.100.199", port = config.mobilePort }
+}
+
+local appParams = {
+  [1] = { appName = "server", appID = "0001", fullAppID = "SPT" },
+  [2] = { appName = "server", appID = "0001", fullAppID = "SPT" }
+}
+
+local addCommandParams = {
+  { cmdID  = 001, menuParams = {menuName = "menu_1"}},
+  { cmdID  = 002, menuParams = {menuName = "menu_2"}},
+  { cmdID  = 003, menuParams = {menuName = "menu_3"}}
+}
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
```

Comment:
This step causes the INI file to be corrupted(The `KeyPath` and `CertificatePath` are not being reset correctly)

Reply:
@ShobhitAd As you pointed, mentioned step causes the INI file to be changed for test purposes. But values of all parameters of INI file are reverted back to its initial state by step 
`runner.Step("Stop SDL, restore SDL settings", common.postconditions)` 

At the begining of the script function `common.preconditions()` uses `actions.sdl.setSDLIniParameter()` for modification of INI file, which store all modifications internally.
 At the end of the script function `common.postcondition` uses `actions.sdl.restoreSDLIniParameters()` for restore all modified parameters to its initial values.

Reply:
Please refer to https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2202#discussion_r314042472

Reply:
INI corruption issue still exists. Please refer to https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2202#discussion_r314763689